examples/design/design_shapes.py

```diff
@@ -59,8 +74,34 @@ def f_objective(self, x, grad):
         return length
 
 
-# Set up the configuration for the design.
-
+# %%[markdown]
+# ## Configuring the Design
+#
+# The Design is configured by passing in a dictionary that provides the build stages to
+# be run, and the way that each of those build stages should be set up. In particular
+# in this case we define the following:
+#
+# - The Plasma build stage:
+#   - uses the `MakeParameterisedShape` Builder class.
+#   - parameterises the plasma shape using the `JohnerLCFS` GeometryParameterisation
+#     class.
+#   - maps the R_0 and A Design parameters to the r_0 and a shape parameters.
+#   - labels the resulting component as "Shape".
+#
+# - The TF Coils build stage:
+#   - uses the `MakeOptimisedShape` Builder class.
+#   - parameterises the TF Coils centerline shape using the `PrincetonD`
+#     GeometryParameterisation class.
+#   - maps the r_tf_in_centre and r_tf_out_centre Design parameters to the x1 and x2
+#     shape parameters. The dz shape parameter is set to 0 and fixed. The x1 shape
```

Comment:
Is there currently a way to view the optimisation variables without diving into the code? 
From a user perspective it would probably be nice to be either able to print that out or maybe better have a diagram explaining what the variables are in the docs (but that doesnt really work for all classes).

If not I think we should say something like the optimisation variables for PrinectonD are <vars> which modify <eg the z offset of the coil>

Reply:
I was thinking that for geometry parameterisation classes we could implement a function that allows the default settings of the class to be drawn with some annotations, then use a custom sphinx directive to automatically generate and include the diagram in our docs. That should be possible, so I'll put an issue in. I agree that it would be useful to be able to print the parameters used by a parameterisation, regardless of that auto-plotting, so I'll put another issue in for that.

Reply:
See #607 #608